File: sudoku_net/sudokuTester/SolverTest.cs
using sudokuNet;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using Xunit;

// example taken from https://www.websudoku.com/
namespace sudokuTester {
    public class SolverTest {
        private readonly string testPath = @"D:\Users\frozen\Documents\03_programming\online\sudokuNet\sudokuTester\tests\";

        [Fact]
        public void TestEasy() {

            var rows = File.ReadLines(testPath + "easy1_input.txt").ToList();
            Board board = new Board(rows);
            Solver solver = new Solver();
            Board solved = solver.Solve(board);

            string expected = File.ReadAllText(testPath + "easy1_solution.txt");
            Assert.Equal(expected, solved.ToString());
        }

        [Fact]
        public void TestEvil() {
            var rows = File.ReadLines(testPath + "evil1_input.txt").ToList();
            Board board = new Board(rows);
            Solver solver = new Solver();
            Board solved = solver.Solve(board);
            Debug.Write(solved.ToString());

            string expected = File.ReadAllText(testPath + "evil1_solution.txt");
            Assert.Equal(expected, solved.ToString());
        }


        [Fact]
        public void TestHardest() {
            var rows = File.ReadLines(testPath + "hardest_input.txt").ToList();
            Board board = new Board(rows);
            Solver solver = new Solver();
            Board solved = solver.Solve(board);

            string expected = File.ReadAllText(testPath + "hardest_solution.txt");
            Assert.Equal(expected, solved.ToString());
        }



    }
}

File: sudoku_java/src/tests/java/SolverTest.java
package tests.java;

import main.java.Board;
import main.java.Solver;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.io.File;
import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.file.Files;


public class SolverTest {
    private final String testPath = "D:/Users/frozen/Documents/03_programming/online/sudoku/sudoku_java/src/tests/java/tests/";

    @Test
    public void TestEasy() throws IOException {
        var rows = Files.readAllLines(new File(testPath + "easy1_input.txt").toPath(), Charset.defaultCharset());
        Board board = new Board(rows);
        Solver solver = new Solver();
        Board solved = solver.Solve(board);

        String expected = Files.readString(new File(testPath + "easy1_solution.txt").toPath(), Charset.defaultCharset());
        assertEquals(expected, solved.toString());
    }

    @Test
    public void TestEvil() throws IOException {
        var rows = Files.readAllLines(new File(testPath + "evil1_input.txt").toPath(), Charset.defaultCharset());
        Board board = new Board(rows);
        Solver solver = new Solver();
        Board solved = solver.Solve(board);

        String expected = Files.readString(new File(testPath + "evil1_solution.txt").toPath(), Charset.defaultCharset());
        assertEquals(expected, solved.toString());
    }


    @Test
    public void TestHardest() throws IOException {
        var rows = Files.readAllLines(new File(testPath + "hardest_input.txt").toPath(), Charset.defaultCharset());
        Board board = new Board(rows);
        Solver solver = new Solver();
        Board solved = solver.Solve(board);

        String expected = Files.readString(new File(testPath + "hardest_solution.txt").toPath(), Charset.defaultCharset());
        assertEquals(expected, solved.toString());
    }


}

